src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -316,7 +316,15 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 		}
 
 		// Pass the sequence along to the capability
-		const [command, ...args] = data.split(';');
+		const dataArr = data.split(';');
+		if (!dataArr.length) {
+			return false;
+		}
+		const command = dataArr[0];
+		const args: string[] | undefined = dataArr.length > 1 ? dataArr.slice(1) : undefined;
+		if (args && args?.length > 2) {
+			return false;
+		}
```

Comment:
The number of args are variable, > 2 is possible. The strict check was only for the CommandLine case

Reply:
yes, I had noticed and fixed that already 👍🏼